### PR TITLE
detect: free all tenant detect engines

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #10219 

Free all tenants registered in the master.

(cherry picked from commit a4d80bc7c4910170aba950db0a497124712b330a)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6549](https://redmine.openinfosecfoundation.org/issues/6549)

Describe changes:
- Removed debug-related changes from CI pipelines

Updates:
- Added --debug-failed to suricata-verify tests.

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=pr/1600
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on ASAN_TLPR1_cfg.
  

Pipeline 18067

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on ASAN_TLPR1_cfg.
  

Pipeline 18077

File: src/detect-engine-loader.c [hunk 1]
```diff
@@ -77,23 +77,11 @@ char *DetectLoadCompleteSigPath(const DetectEngineCtx *de_ctx, const char *sig_f
 
     /* Path not specified */
     if (PathIsRelative(sig_file)) {
-        if (ConfGet(varname, &defaultpath) == 1) {
```
Comment (victorjulien) [inline src/detect-engine-loader.c:80]: should this be removed?

Comment (victorjulien) [reply src/detect-engine-loader.c:80]: doesn't look like defaultpath is initialized to anything other than NULL

Comment (jlucovsky) [reply src/detect-engine-loader.c:80]: This commit is needed: 39570d8d0500a4a5fe8a9e9bc1aef406e8a7d4f4

**Comment by jlucovsky** [comment]
Continued in #10313

**Pull request closed without merging**